core/src/from_meta.rs

```diff
@@ -246,6 +246,47 @@ impl<T: syn::parse::Parse, P: syn::parse::Parse> FromMeta for syn::punctuated::P
     }
 }
 
+/// Parsing support for an array, i.e. "\[1 + 2, 2 - 2, 3 * 4\]".
+impl FromMeta for syn::ExprArray {
+    fn from_value(value: &Lit) -> Result<Self> {
+        if let Lit::Str(ref ident) = *value {
+            ident
+                .parse::<syn::ExprArray>()
+                .map_err(|_| Error::unknown_lit_str_value(ident))
+        } else {
+            Err(Error::unexpected_lit_type(value))
+        }
+    }
+}
+
+macro_rules! from_numeric_array {
+    ($ty:ident) => {
+        impl FromMeta for Vec<$ty> {
```

Comment:
An example baked into the macro would be good too. It doesn't need to run - illustrating how it can be used like you did for the `ExprArray` would be sufficient.